app/views/documents_v2/launch.html.haml

```diff
@@ -0,0 +1,158 @@
+-if @document
+  - if @document.shared
+    .row
+      .small-12.text-center.columns.end
+        #loading-message
+          %span#loading-image
+            =image_tag "loading.gif"
+          %span#loading-text
+            Loading...
+        = link_to h(@button_text), @button_url, class: :"launch-button", target: "_blank"
+    .row.more-space-above
+      .small-12.text-center.columns
+        = link_to 'Reset to Original', '#', class: :"original-reset"
+      #launch-reset-message.small-12.text-center.columns{style: "display: none;"}
+  -else
+    .row
+      .small-12.text-center.columns{style: "color: #990000" }
+        Error: The requested document is not shared.
+-else
+  .row
+    .small-12.text-center.columns{style: "color: #990000" }
+      Error: The requested document could not be found.
+
+-if @document && @document.shared
+  :javascript
+    $(document).ready(function() {
+      if (#{@in_a_window.to_s}) {
+        // Set all links to replace this page instead of opening yet another window.
+        $('a').attr('target', '');
+        return;
+      }
+
+      // Hide the launch buttons until we talk with LARA
+      $('.launch-button, .original-reset').hide();
+
+      // But show them after a while, just in case
+      var showTimeoutId = setTimeout(function() {
+        $('#loading-text').html("Still loading!  You may want to reload this page to try again.")
+      }, 10000);
+
+      var resetUrlTemplate = '#{@reset_url}',
+          phone = iframePhone.getIFrameEndpoint(),
+          interactiveRunState, extractError, haveResetInfo, interactiveStateUrl;
+
+      extractError = function (jqXHR) {
+        var error = '', result;
+        try {
+          result = JSON.parse(jqXHR.responseText);
+          if (result && result.errors) {
+            error = result.errors.join('. ');
+            if (error.length > 0) {
+              error = "The error returned was: <em>" + error + "</em>";
+            }
+          }
+        } catch (e) {}
+        return error;
+      };
+
+      phone.addListener('initInteractive', function (interactiveData) {
+        var launchParams = {url: interactiveData.interactiveStateUrl, source: #{@document.id}};
+
+        interactiveRunState = interactiveData.interactiveState || {};
+        interactiveStateUrl = interactiveData.interactiveStateUrl;
+
+        clearTimeout(showTimeoutId);
+
+        // update the launch button to add the interactive run state api url
+        $('.launch-button').each(function () {
+          var $this = $(this),
+              launchUrl = $this.attr('href');
+          $this.attr('href', $.param.querystring(launchUrl, {launchFromLara: Base64.encode(JSON.stringify(launchParams))}));
+        });
+
+        phone.addListener('getInteractiveState', function () {
+          phone.post('interactiveState', 'nochange');
+        });
+
+        $('#loading-message').hide();
+        $('.launch-button,.original-reset').show();
```

Comment:
In the meeting today, we decided to now show the reset button on init. But instead it should only be shown after launch button is pressed, or if the interactiveData has a learner document.